packages/syft/src/syft/service/action/action_graph.py

```diff
@@ -79,7 +78,7 @@ def make_created_at(cls, v: Optional[DateTime]) -> DateTime:
         return DateTime.now() if v is None else v
 
     @staticmethod
-    def from_action(action: Action, credentials: SyftVerifyKey):
+    def from_action(action: Action, credentials: SyftVerifyKey) -> "NodeActionData":
```

Comment:
This should be a `classmethod` returning `Self`. `return cls(...` instead of `return NodeActionData(...`.

Reply:
Thanks @kiendang. Done!